### PR TITLE
Move non-init code out of __init__.py files

### LINKED ISSUE #378: Make __init__.py small, having only necessary stuffs
The current __init__.py in many submodules are very large and make it difficult to keep track of what will be automatically loaded for the submodule.

Should move implementations to separate file and keep __init__.py to list only function names when appropriate.

### PR DESCRIPTION
**Pull request by bact**
This PR will close #378

- Move non-init code out of `__init__.py` in submodules
- Move `Trie` class from `pythainlp.tokenize` to `pythainlp.util`
- Refactor few codes related to `Trie`
- Add few more test cases for `dict_trie()`

### CODE REVIEW

File: pythainlp/spell/__init__.py [hunk 1]
```diff
@@ -3,88 +3,14 @@
 Spell checking
 """
 
-from typing import List
+from .pn import NorvigSpellChecker
+from .spell import correct, spell
 
-from .pn import DEFAULT_SPELL_CHECKER, NorvigSpellChecker
+DEFAULT_SPELL_CHECKER = NorvigSpellChecker()
```
Comment (p16i) [inline pythainlp/spell/__init__.py]: I'm curious how the instantiation works here. Do we always have this spell checker object when importing `pythainlp`? In other word, when the spell check is instantiated. Can we do lazy instantiation here, i.e. only creating the object when needed?

Comment (bact) [reply pythainlp/spell/__init__.py]: Currently the instance is created immediately at the module loading time. But, yes, lazy instantiation would be better.

**Comment by wannaphong** [comment]
I update docs.

**Review by wannaphong — APPROVED** [review]
It looks very nice.

**Pull request merged**